### PR TITLE
Added function to unregister packets on plugin disable. [Dev System Only]

### PR DESCRIPTION
**Pull request by alexdev03**
While testing and updating Velocitab, restarting the proxy could take a while. With this pr I unregister packets when the plugin disable, so with a plugin manager it's possibile to register team packets again.

### CODE REVIEW

File: src/main/java/net/william278/velocitab/packet/ScoreboardManager.java [hunk 1]
```diff
@@ -108,12 +109,20 @@ public void registerPacket() {
...
+                    .mapping(0x5A, MINECRAFT_1_19_4, false);
+            packetRegistration.register();
         } catch (Throwable e) {
             plugin.log(Level.ERROR, "Failed to register UpdateTeamsPacket", e);
         }
     }
 
+    public void unregisterPacket() {
+        try {
+            packetRegistration.unregister();
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/packet/ScoreboardManager.java:121]: can we null-check this in the event the plugin terminates before registration? just a little guard clause would probably be fine.

File: src/main/java/net/william278/velocitab/packet/PacketRegistration.java [hunk 2]
```diff
@@ -72,16 +78,48 @@ public PacketRegistration<P> mapping(
...
+            }
+        }
+
+        @SuppressWarnings("unchecked")
+        public void unregister() {
+            try {
+
+                final StateRegistry.PacketRegistry packetRegistry = direction == ProtocolUtils.Direction.CLIENTBOUND
+                        ? (StateRegistry.PacketRegistry) STATE_REGISTRY$clientBound.invoke(stateRegistry)
+                        : (StateRegistry.PacketRegistry) STATE_REGISTRY$serverBound.invoke(stateRegistry);
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/packet/PacketRegistration.java:105]:
```suggestion
            try {
                final StateRegistry.PacketRegistry packetRegistry = direction == ProtocolUtils.Direction.CLIENTBOUND
                        ? (StateRegistry.PacketRegistry) STATE_REGISTRY$clientBound.invoke(stateRegistry)
                        : (StateRegistry.PacketRegistry) STATE_REGISTRY$serverBound.invoke(stateRegistry);
```
style nit: just removing this newline here

File: src/main/java/net/william278/velocitab/packet/PacketRegistration.java [hunk 3]
```diff
@@ -72,16 +78,48 @@ public PacketRegistration<P> mapping(
         }
 
         public void register() {
+
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/packet/PacketRegistration.java:81]: style nit: probably don't need a newline here.

**Review by WiIIiam278 — CHANGES_REQUESTED** [review]
Seems sensible enough. Not that I particularly want to _support_ people reloading with plugin managers, I do get the rationale. We should tidy up after ourselves :). Paging @4drian3d for thoughts on this?

A few nitpicks either way:

File: src/main/java/net/william278/velocitab/Velocitab.java [hunk 4]
```diff
@@ -148,6 +155,12 @@ private void prepareScoreboardManager() {
         }
     }
 
+    private void disableScoreboardManager() {
+        if(scoreboardManager!=null && settings.isSortPlayers()) {
```
Comment (WiIIiam278) [inline src/main/java/net/william278/velocitab/Velocitab.java:159]:
```suggestion
if (scoreboardManager !=null && settings.isSortPlayers()) {
```

**Comment by 4drian3d** [comment]
> Seems sensible enough. Not that I particularly want to _support_ people reloading with plugin managers, I do get the rationale. We should tidy up after ourselves :). Paging @4drian3d for thoughts on this?
> 
> A few nitpicks either way:

Velocity does not have a plugin status enabled or disabled.
Although, with ServerUtils you can unload a plugin in runtime, this would not be ideal and I would prefer to avoid it, although this pull request would solve all the problems that could arise in case someone does it

**Review by WiIIiam278 — APPROVED** [review]

**Pull request merged**